rust/template/differential_datalog/src/program/mod.rs

```diff
@@ -85,7 +89,111 @@ type TKeyAgent<S> = TraceAgent<KeyTrace<S>>;
 type TValEnter<P, T> = TraceEnter<TValAgent<P>, T>;
 type TKeyEnter<P, T> = TraceEnter<TKeyAgent<P>, T>;
 
-/// Diff associated with records in differential dataflow
+#[derive(
+    Abomonation, Copy, Ord, PartialOrd, Eq, PartialEq, Hash, Debug, Clone, Serialize, Deserialize,
+)]
+pub struct CheckedWeight {
+    pub value: i32,
+}
+
+/// CheckedWeight trait implementations
+
+impl Semigroup for CheckedWeight {
+    fn is_zero(&self) -> bool {
+        self.value == 0
+    }
+}
+
+impl<'a> AddAssign<&'a Self> for CheckedWeight {
+    fn add_assign(&mut self, other: &'a Self) {
+        // intentional panic on overflow
+        self.value = self
+            .value
+            .checked_add(other.value)
+            .expect("Weight overflow");
+    }
+}
+
+impl Add for CheckedWeight {
+    type Output = Self;
+
+    fn add(self, other: Self) -> Self {
+        // intentional panic on overflow
+        CheckedWeight {
+            value: self
+                .value
+                .checked_add(other.value)
+                .expect("Weight overflow"),
+        }
+    }
+}
+
+impl Mul for CheckedWeight {
+    type Output = Self;
+    fn mul(self, rhs: Self) -> Self::Output {
+        // intentional panic on overflow
+        CheckedWeight {
```

Comment:
```suggestion
        Self {
```

Reply:
I actually find the original version easier to read. @Kixiron , so you think `Self` is more idiomatic?
In any case, @mbudiu-vmw , since you made this change everywhere you probably want it here as well.

Reply:
Yah, `Self` is more idiomatic and it's more resistant to annoyances while refactoring since it just refers to the current struct, one less thing to change